tests/utils/test_train_utils.py

```diff
@@ -157,6 +158,27 @@ def test_update_confidence_type(
     assert component_config[MODEL_CONFIDENCE] == model_confidence
 
 
+@pytest.mark.parametrize(
+    "component_config, raises_exception",
+    [
+        ({TOLERANCE: 0.5}, False),
+        ({TOLERANCE: 0.0}, False),
+        ({TOLERANCE: 1.0}, False),
+        ({TOLERANCE: -1.0}, True),
+        ({TOLERANCE: 2.0}, True),
+        ({}, False),
+    ],
+)
+def test_tolerance_setting(
+    component_config: Dict[Text, float], raises_exception: bool,
+):
+    if raises_exception:
+        with pytest.raises(InvalidConfigException):
+            train_utils._check_tolerance_setting(component_config)
+    else:
+        train_utils._check_tolerance_setting(component_config)
```

Comment:
That will result in key errors because other functions in `validate_configuration_settings` aren't written in a way such that they can handle certain keys not existing in `component_config` passed to them.

Reply:
I see... Thanks, Daksh. I guess we could still speculate about other options but I don't think it's something for now.